x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/factories/utils/strip_non_ecs_fields.ts

```diff
@@ -118,6 +119,11 @@ const computeIsEcsCompliant = (value: SourceField, path: string) => {
   const ecsField = ecsFieldMap[path as keyof typeof ecsFieldMap];
   const isEcsFieldObject = getIsEcsFieldObject(path);
 
+  // validate if value is a keyword type
+  if (ecsField?.type === 'keyword') {
```

Comment:
I added update mappings in `x-pack/plugins/security_solution/common/field_maps/8.0.0/alerts.ts`

We might not need this check at all, if the rest of mappings will be fixed on alerting framework side: https://github.com/elastic/kibana/issues/187768#issuecomment-2217418380